src/main/scala/io/github/mtsongithub/doetl/sparkdialectextensions/clickhouse/spark35/ClickhouseDialectExtension.scala

```diff
@@ -165,14 +166,42 @@ private object ClickhouseDialectExtension extends JdbcDialect {
     case TimestampType =>
       logger.debug("Custom mapping applied: Datetime64(6) for 'TimestampType'")
       Some(JdbcType("Datetime64(6)", Types.TIMESTAMP))
-    case ArrayType(et, _) =>
+    case ArrayType(et, nullable) =>
       logger.debug("Custom mapping applied: Array[T] for ArrayType(T)")
       getJDBCType(et)
         .orElse(JdbcUtils.getCommonJDBCType(et))
-        .map(jdbcType => JdbcType(s"Array(${jdbcType.databaseTypeDefinition})", Types.ARRAY))
+        .map(jdbcType =>
+          if (nullable) {
+            logger.debug(
+              s"Mapping to Nullable type for Array: Nullable(${jdbcType.databaseTypeDefinition})")
+            JdbcType(s"Array(Nullable(${jdbcType.databaseTypeDefinition}))", Types.ARRAY)
+          } else
+            JdbcType(s"Array(${jdbcType.databaseTypeDefinition})", Types.ARRAY))
     case _ =>
       logger.debug(
         s"No custom JDBC type mapping for DataType: ${dt.simpleString}, default driver mapping is used")
       None
   }
+
+  override def createTable(
+      statement: Statement,
+      tableName: String,
+      strSchema: String,
+      options: JdbcOptionsInWrite): Unit = {
+    statement.executeUpdate(s"CREATE TABLE $tableName (${parseColumnDefinitions(strSchema)}) ${options.createTableOptions}")
+  }
+
+  private def parseColumnDefinitions(columnDefinitions: String): String = {
```

Comment:
Let's add a comment with links to Spark JdbcUtils class sources:
* https://github.com/apache/spark/blob/branch-3.5/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala#L919-L923
* https://github.com/apache/spark/blob/branch-3.5/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala#L823-L824

This is the reason why we should parse raw SQL in the first place